DOCS-3280 Adds Frustration Signals Documentation (#14119)
DataDog/documentation: content/en/real_user_monitoring/frustration_signals/_index.md

  text: 'Frustration Signals Dashboard'
- link: '/real_user_monitoring/explorer'
  tag: 'Documentation'
  text: 'Learn about the RUM Explorer'
- link: '/real_user_monitoring/session_replay'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
---

## Overview

<div class="alert alert-warning">
For access to frustration signals, contact <a href="/help">Datadog Support</a>.
</div>

Frustration signals identify the highest points of user friction in your application by surfacing moments where users exhibit frustration. 

RUM collects three types of frustration signals:

Rage Clicks
: A user clicks on an element more than three times in a one-second rolling window.

Dead Clicks
: A user clicks on a static element thinking it is interactive.

Error Clicks
: A user clicks on an element right before a JavaScript error occurs.

## Requirements

First, you need the Browser RUM SDK version >= 4.9.0.

To start collecting frustration signals, add the following to your SDK configuration:

```
DD_RUM.init({
  trackInteractions: true,
  trackFrustrations: true
})
```

Frustration signals require actions. Enabling `trackFrustrations` automatically enables `trackInteractions`.

## Usage

> Frustration signals appear as high-level data points representing sources of user frustration on the [**RUM Applications** page][1]. The [RUM Explorer][2] displays a count of frustration signals as a column in the **List** visualization. 

Reviewer: not sure where the second sentence came from, we can remove. it's not true unless a user does this themselves

Author: ```suggestion
Frustration signals appear as high-level data points representing sources of user frustration on the [**RUM Applications** page][1]. To display a list of frustration signals count in the [RUM Explorer][2], click the **Options** button and add a column with `@session.frustration.count`. 
```

### Application list

Hover over the list of browser sessions and click on a session to observe a user's frustrated click behavior. Or, click **Frustrated Sessions** to access sessions with a frustration signal. 

### Explore the frustration signals dashboard

The **Frustration Signals** dashboard provides an overview of frustration levels across your application, showing you topics like the most frustrated users and pages with the highest amount of frustration signals.

> 

Reviewer: this is where we should add the dashboard, can you use the image i sent over yesterday? that one is what the final will look like

Author: I included the image in the RUM Frustration Signals Dashboard page already, I don't think we should duplicate the image in two places.

### Search for frustration signals

Search through all the data collected by RUM in the [RUM Explorer][4] to surface trends on frustration signals, analyze patterns with greater context, or export them into [dashboards][5] and [monitors][6].

Enter a facet in the search query to start your search. Available search fields include:

Frustration Type
: Find actions with frustration signals. For example, if you want to see any actions that had a rage click, add `action.frustration_type:rage_click` to the search query.

Frustration Count
: Find sessions and views where any frustration signal occurred. For example, if you want to find any user sessions or views that had at least one frustration signal occur, add `session.frustration.count:>0` or `view.frustration.count:>0` to the search query.

#### Sessions

Click on a session with a value in the **Frustration Count** column to examine the user frustration detected. You can see the type of signal (ranging from `rage click`, `dead click`, or `error click`) and the event timeline that shows what occurred during the session.

#### Views

> Click on a view to identify if a user was frustrated with the `frustration detected` tag.

Reviewer: suggesting to reword "Click on a view to identify if a user was frustrated on a specific page with the 'frustration detected' tag

{{< img src="real_user_monitoring/frustration_signals/frustration_signals_in_performance_tab.png" alt="Frustration Signals Actions in the Events dropdown menu in the performance waterfall graph" style="width:90%;" >}}

The action waterfall contains actions with frustration signals.

{{< img src="real_user_monitoring/frustration_signals/actions_frustration_signal.png" alt="Frustration signals detected as an action" style="width:90%;" >}}

#### Actions

The Actions tab displays the `frustration detected` tag if the selected action contains a frustration signal.

If multiple frustration signals occur in an action, they are displayed under **What Happened** in the action panel.

{{< img src="real_user_monitoring/frustration_signals/actions_panel_multiple_frustration_signals.png" alt="Multiple frustration signal types detected in the action under What Happened" style="width:90%;" >}}

#### Errors

> Click on an error in the **Errors** tab to open a side panel with error details. Under the **Parent hierarchy**, you can see if a frustration signal occurred next to the action.

Author: ```suggestion
Click on an error in the **Errors** tab to open a side panel with error details. You can see if a frustration signal occurred.
```

{{< img src="real_user_monitoring/frustration_signals/errors_tab.png" alt="Errors Tab in the Actions side panel" style="width:90%;" >}}

## Watch frustration signals in Session Replay

> In [Session Replay][7], you can observe a replication of real user activity. Replays provide video evidence of the actions users take when they exhibit signs of frustration.

Reviewer: add "video-like" before replication

A session replay's user journey details the events that occur in chronological order. Hover over an event to move to that point in time in the replay, for example, when a dead click occurred.

{{< img src="real_user_monitoring/frustration_signals/session_replay_frustration_signals.png" alt="Frustration signals appear in a browser recording" style="width:90%;" >}}

 For more information, see the [Session Replay documentation][8].

## Create alerts for frustration signals

You can create monitors and set alerts on frustration signals to notify you or your team if any frustration signals occur on an important page in your application.

> 

Author: ```suggestion

For example, to set an alert that notifies you if any frustration signals occur on a specific page:

```

---
title: Frustration Signals
kind: documentation
further_reading:
- link: '/real_user_monitoring/dashboards/frustration_signals_dashboard/'
  tag: 'Documentation'
  text: 'Frustration Signals Dashboard'
- link: '/real_user_monitoring/explorer'
  tag: 'Documentation'
  text: 'Learn about the RUM Explorer'
- link: '/real_user_monitoring/session_replay'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
---

## Overview

<div class="alert alert-warning">
For access to frustration signals, contact <a href="/help">Datadog Support</a>.
</div>

> Frustration signals identify the highest points of user friction in your application by surfacing moments where users exhibit frustration. 

Reviewer 2: ```suggestion
Frustration signals identify the highest points of user friction in your application by surfacing moments when users exhibit frustration. 
```

RUM collects three types of frustration signals:

Rage Clicks

> : A user clicks on an element more than three times in a one-second rolling window.

Reviewer 2: ```suggestion
: A user clicks on an element more than three times in a one-second sliding window.
```

they mean the same thing, but the other one said "sliding" so let's stick with sliding

Frustration Count
: Find sessions and views where any frustration signal occurred. For example, if you want to find any user sessions or views that had at least one frustration signal occur, add `session.frustration.count:>1` or `view.frustration.count:>1` to the search query.

#### Sessions

Click on a session with a value in the **Frustration Count** column to examine the user frustration detected. You can see the type of signal (ranging from `rage click`, `dead click`, or `error click`) and the event timeline that shows what occurred during the session.

#### Views

Click on a view to identify if a user was frustrated on a specific page with the `frustration detected` tag.

{{< img src="real_user_monitoring/frustration_signals/frustration_signals_in_performance_tab.png" alt="Frustration Signals Actions in the Events dropdown menu in the performance waterfall graph" style="width:90%;" >}}

The performance waterfall displays actions containing frustration signals.

{{< img src="real_user_monitoring/frustration_signals/actions_frustration_signal.png" alt="Frustration signals detected as an action" style="width:90%;" >}}

#### Actions

> The Actions tab displays the `frustration detected` tag if the selected action contains a frustration signal.

Reviewer 2: ```suggestion
The **Actions** tab displays the `frustration detected` tag if the selected action contains a frustration signal.
```

If multiple frustration signals occur in an action, they are displayed under **What Happened** in the action panel.

{{< img src="real_user_monitoring/frustration_signals/actions_panel_multiple_frustration_signals.png" alt="Multiple frustration signal types detected in the action under What Happened" style="width:90%;" >}}

#### Errors

Click on an error in the **Errors** tab to open a side panel with error details. You can see if a frustration signal occurred.

{{< img src="real_user_monitoring/frustration_signals/errors_tab.png" alt="Errors Tab in the Actions side panel" style="width:90%;" >}}

## Watch frustration signals in Session Replay

In [Session Replay][7], you can observe a video-like replication of real user activity. Replays provide video evidence of the actions users take when they exhibit signs of frustration.

> A session replay's user journey details the events that occur in chronological order. Hover over an event to move to that point in time in the replay, for example, when a dead click occurred.

Reviewer 2: ```suggestion
A session replay's user journey details the events that occur in chronological order. Hover over an event to move to that point in time in the replay: for example, when a dead click occurred.
```

{{< img src="real_user_monitoring/frustration_signals/session_replay_frustration_signals.png" alt="Frustration signals appear in a browser recording" style="width:90%;" >}}

 For more information, see the [Session Replay documentation][8].

## Create alerts for frustration signals

You can create monitors and set alerts on frustration signals to notify you or your team if any frustration signals occur on an important page in your application.

For example, to set an alert that notifies you if any frustration signals occur on a specific page:

{{< img src="real_user_monitoring/frustration_signals/rum_monitor_frustration_count.png" alt="Create a RUM Monitor that alerts on the count of frustration signals" style="width:90%;" >}}

For more information, see the [Real User Monitoring Monitor documentation][9].

## Troubleshooting

### Why is a rage click not created when a user presses a key (like Delete) on the keyboard?

Frustration signals are generated from mouse clicks, not keyboard strokes. 

> ### Why does the side panel show a session has a different number of frustration signals that in the event timeline?

Reviewer 2: ```suggestion
### Why does the side panel show that a session has a different number of frustration signals than in the event timeline?
```

> If a session is live, it is fetching information, and may cause the banners to reflect a different number than those in the timeline.

Reviewer 2: ```suggestion
If a session is live, it is fetching information and may cause the banners to reflect a different number than those in the timeline.
```
